components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/CacheBackedApplicationDAO.java

```diff
@@ -499,6 +523,20 @@ public boolean isApplicationDiscoverable(String resourceId, String tenantDomain)
         return appDAO.isApplicationDiscoverable(resourceId, tenantDomain);
     }
 
+    @Override
+    public List<TrustedApp> getTrustedApps(PlatformType platformType) throws IdentityApplicationManagementException {
+
+        List<TrustedApp> trustedApps = getTrustedAppsFromCacheByPlatformType(platformType);
+        if (trustedApps == null) {
+            // Cache miss, fetch from DB.
```

Comment:
Lets add a debug log here

Reply:
The getTrustedAppsFromCacheByPlatformType method already has 2 debug logs for cache hit and cache miss scenarios